### DISCUSSION TITLE
Python documentation

### DISCUSSION THREAD

**Discussion by jhultgre** [Autodesk/maya-usd#1022] [Q&A]
Is there any documentation anywhere on how I can interact with a usd stage in maya from my own python tools?

**Comment by TechArtistG** [comment]
Not that I know of.  Best way is to crack open some of the test scripts in the repo.
You can however start a doc server from Maya and see all the modules and members loaded:
Py3:
```
# Maya 2021/Python 3.x doc server
# After execution go to http://localhost:8089
import pydoc
import threading

def threadFn():
    pydoc._start_server(pydoc._url_handler, "localhost", 8089)
    
th = threading.Thread(target=threadFn)
th.start()
```

Py2:
```
import thread
import pydoc

def startPyDocServer(port=8089):
    # Calling pydoc.serve is blocking so can't call it directly
    thread.start_new_thread(pydoc.serve, (port,))

startPyDocServer()
```

**Comment by kxl-adsk** [comment]
It's still on our list of things to do. So the best answer I can give is to look at python tests (especially UFE related). They will give you the best examples of how to work with proxy nodes and access the stage. One starting point could be https://github.com/Autodesk/maya-usd/blob/19d82e86bfc31a6adf2843a382d531191519ed8f/test/testUtils/mayaUtils.py#L190

As well, don't hesitate to ask here for specific things you are trying to do.
